presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PushdownSubfields.java

```diff
@@ -529,47 +545,60 @@ private static NestedField nestedField(String name)
         }
 
         private static final class SubfieldExtractor
-                extends DefaultExpressionTraversalVisitor<Void, Context>
+                extends DefaultRowExpressionTraversalVisitor<Context>
         {
-            private final TypeProvider typeProvider;
+            private final StandardFunctionResolution functionResolution;
+            private final ExpressionOptimizer expressionOptimizer;
+            private final ConnectorSession connectorSession;
 
-            private SubfieldExtractor(TypeProvider typeProvider)
+            private SubfieldExtractor(StandardFunctionResolution functionResolution, ExpressionOptimizer expressionOptimizer, ConnectorSession connectorSession)
```

Comment:
This class has also been modified significantly since we do not have such fine granularity visits for RowExpressions. Hence, on "default" visits, we simply visit the RowExpression's children.